engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/source/KafkaSourceFactory.scala

```diff
@@ -42,10 +42,26 @@ class KafkaSourceFactory[K: ClassTag, V: ClassTag](deserializationSchemaFactory:
                                                    timestampAssigner: Option[TimestampWatermarkHandler[ConsumerRecord[K, V]]],
                                                    formatterFactory: RecordFormatterFactory,
                                                    processObjectDependencies: ProcessObjectDependencies)
-  extends FlinkSourceFactory[ConsumerRecord[K, V]] with SingleInputGenericNodeTransformation[FlinkSource[ConsumerRecord[K, V]]] with WithCachedTopicsExistenceValidator {
+  extends FlinkSourceFactory[ConsumerRecord[K, V]]
+    with SingleInputGenericNodeTransformation[FlinkSource[ConsumerRecord[K, V]]]
+    with WithCachedTopicsExistenceValidator
+    with WithExplicitPossibleProducedVariableTypes {
 
   protected val topicNameSeparator = ","
 
+  protected val keyTypingResult: TypedClass = Typed.typedClass[K]
```

Comment:
Why you assume there always be typed class?

Reply:
Here K and V are usually a specific class or some generic Map/TypedMap. For specific classes this is the same as previous assumption with Typed[K] and Typed[V]. For generic Map/TypedMap the returning type is overwritten by TypedObjectTypingResult which has explicit fields definition.